### ISSUE TITLE
Nested atomicapp applications error on the same namespace

### ISSUE THREAD

**Issue by charliedrage** [projectatomic/atomicapp#398]
This semi-relates to issue #397

When using:
 - The Docker provider
 - A `atomicapp` container with inherited / nested `atomicapp` application
 - and.. all namespaces are default / not assigned

You encounter an error where you're unable to deploy the second inherited `atomicapp` application, ending up with a `this namespace is already in use` error.

This is due to the fact that you cannot assign a unique namespace to the inherited `atomicapp` container __as well__ as the fact that our current regex expression ` if re.match("%s_+%s+_+[a-zA-Z0-9]{12}" % (self.default_name, self.namespace), container):` within `docker.py` will match the first `atomicapp` container that was deployed.

**Comment by dharmit** [comment]
I hit this issue on `0.2.1`.

**Comment by dustymabe** [comment]
@cdrage I thought we talked about the implementation for this.. If we are deploying two instances of the same application:

* instance of exampleapp (nulecule) - with namespace "default"
    - appa (nulecule)
        + image1
        + appaa (nulecule)
            - image2
    - appb (nulecule)
        + image3
        + image4
* instance of exampleapp (nulecule) - with namespace "default1"
    - appa (nulecule)
        + image1
        + appaa (nulecule)
            - image2
    - appb (nulecule)
        + image3
        + image4

we would create image names:
default_appa_image1_$(RANDOM)
default_appa_appaa_image2_$(RANDOM)
default_appb_image3_$(RANDOM)
default_appb_image4_$(RANDOM)
default1_appa_image1_$(RANDOM)
default1_appa_appaa_image2_$(RANDOM)
default1_appb_image3_$(RANDOM)
default1_appb_image4_$(RANDOM)

or something along those lines

**Comment by cdrage** [comment]
@dustymabe 

Yes, you're right, we should do it along those lines (especially for the Docker provider).

I was a tad confused with https://github.com/projectatomic/nulecule/issues/187 and @rtnpro proposal. But his one has to do more with namespacing in the answers.conf.

The docker provider should detect if it is an inherited application and act accordingly with naming the Docker containers.

**Comment by rtnpro** [comment]
@cdrage ping? we need to sync up on this.

**Comment by cdrage** [comment]
@rtnpro this will be fixed before GA. are you good with @dustymabe 's proposal on the naming scheme?
